src/table/use-column-widths.tsx

```diff
@@ -80,6 +82,36 @@ export function ColumnWidthsProvider({ visibleColumns, resizableColumns, childre
     }
   };
 
+  // Update getContainer when deps change to trigger new observation.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getContainer = useCallback(() => containerRef.current, [resizableColumns, visibleColumns, columnWidths]);
+
+  // Measure wrapper width to determine if the last column can stretch.
+  // When condition changes, updates the last column width style imperatively to avoid having a frame with improper style.
+  const [containerWidth, setContainerWidth] = useState(0);
+  useResizeObserver(getContainer, entry => {
+    if (!resizableColumns) {
+      return;
+    }
+
+    setContainerWidth(entry.contentBoxWidth);
+
+    const lastColumnId = visibleColumns[visibleColumns.length - 1]?.id;
+    const lastCellElement = lastColumnId ? cellsRef.current[lastColumnId] : null;
+    if (!lastCellElement) {
+      return;
+    }
+
+    const totalWidth = visibleColumns.reduce((total, { id }) => total + (columnWidths[id] || DEFAULT_COLUMN_WIDTH), 0);
+    const stretchLastColumn = Boolean(entry.contentBoxWidth > totalWidth);
```

Comment:
What does this `Boolean()` do?

Reply:
In this case - nothing, will remove it. It was copied from the similar line below where the conversion is required to get the correct type.